test/core/ext/filters/event_engine_client_channel_resolver/resolver_fuzzer.cc

```diff
@@ -0,0 +1,265 @@
+                                    absl::string_view /* name */,
+                                    absl::string_view /* default_port */,
+                                    Duration /* timeout */) override {
+      auto finish =
+          [cb = std::move(on_resolve), runner = &engine_->runner_](
+              absl::StatusOr<std::vector<ResolvedAddress>> response) mutable {
+            runner->Run(
+                [cb = std::move(cb), response = std::move(response)]() mutable {
+                  cb(response);
+                });
+            return EventEngine::DNSResolver::LookupTaskHandle::kInvalid;
+          };
+      if (!engine_->hostname_responses_.has_value()) {
+        return finish(engine_->lookup_hostname_response_base_case_);
+      }
+      return finish(*engine_->hostname_responses_);
+    }
+    LookupTaskHandle LookupSRV(LookupSRVCallback on_resolve,
+                               absl::string_view /* name */,
+                               Duration /* timeout */) override {
+      auto finish =
+          [cb = std::move(on_resolve), runner = &engine_->runner_](
+              absl::StatusOr<std::vector<SRVRecord>> response) mutable {
+            runner->Run(
+                [cb = std::move(cb), response = std::move(response)]() mutable {
+                  cb(response);
+                });
+            return EventEngine::DNSResolver::LookupTaskHandle::kInvalid;
+          };
+      if (!engine_->srv_responses_.has_value()) {
+        return finish(engine_->lookup_srv_response_base_case_);
+      }
+      return finish(*engine_->srv_responses_);
+    }
+    LookupTaskHandle LookupTXT(LookupTXTCallback on_resolve,
+                               absl::string_view /* name */,
+                               Duration /* timeout */) override {
+      auto finish =
+          [cb = std::move(on_resolve), runner = &engine_->runner_](
+              absl::StatusOr<std::vector<std::string>> response) mutable {
+            runner->Run(
+                [cb = std::move(cb), response = std::move(response)]() mutable {
+                  cb(response);
+                });
+            return EventEngine::DNSResolver::LookupTaskHandle::kInvalid;
+          };
+      if (!engine_->txt_responses_.has_value()) {
+        return finish(engine_->lookup_txt_response_base_case_);
+      }
+      return finish(*engine_->txt_responses_);
+    }
+    bool CancelLookup(LookupTaskHandle /* handle */) override { return false; }
+
+   private:
+    FuzzingResolverEventEngine* engine_;
+  };
+
+  // members
+  FuzzingEventEngine runner_;
+
+  // responses
+  absl::optional<std::vector<EventEngine::ResolvedAddress>> hostname_responses_;
+  absl::optional<std::vector<EventEngine::DNSResolver::SRVRecord>>
+      srv_responses_;
+  absl::optional<std::vector<std::string>> txt_responses_;
+
+  // base cases
+  const std::string txt_valid_config_ =
+      "grpc_config=[{\"serviceConfig\":{\"loadBalancingPolicy\":\"round_"
```

Comment:
I think this kind of input should be generated by the fuzzer, so that it can test that we correctly handle various types of malformed input.

Reply:
Having a "this is good/this is bad" example for the fuzzer is still useful though: it allows additional checks to be made if the fuzzer understands that this must definitely pass or fail.

I'd probably encourage instead moving the proto to something like:

```
oneof {
  TXTRecordType enumerated_value = 1;
  string arbitrary_value = 2;
}
```

with a weaker check on `arbitrary_value` than is possible with `enumerated_value`

Reply:
Are you suggesting that `arbitrary_value` here would be the TXT record content if the enumerated value was "invalid"? If so, cool. If not, I may do that anyway.

Reply:
Oh, I missed the `oneof` ... that must be what you meant. SG.